test/unit/metrics/test_metricframe_streaming.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+import pytest
+import sklearn.metrics as skm
+
+import fairlearn.metrics as metrics
+from test.unit.input_convertors import conversions_for_1d
+from .data_for_test import y_t, y_p, g_4
+from .utils import batchify
+
+
+@pytest.mark.parametrize("transform_y_p", conversions_for_1d)
+@pytest.mark.parametrize("transform_y_t", conversions_for_1d)
+@pytest.mark.parametrize("num_batches", [1, 3, 5])
+def test_sample_params(transform_y_t, transform_y_p, num_batches):
+    sample_weights = np.random.rand(len(y_t))
+    batch_generator = batchify(num_batches, y_t, y_p, g_4, sample_weights)
+    target = metrics.MetricFrame(skm.recall_score, [], [], sensitive_features=[],
+                                 sample_params=None, streaming=True)
+
+    with pytest.raises(ValueError) as excinfo:
+        _ = target.control_levels
+    assert "No data to process" in str(excinfo.value)
+
+    for y_t_batches, y_p_batches, g_4_batches, sw_batches in batch_generator:
+        g_f = pd.DataFrame(data=g_4_batches, columns=['My feature'])
+        target.add_data(transform_y_t(y_t_batches),
+                        transform_y_p(y_p_batches),
+                        sensitive_features=g_f,
+                        sample_params={'sample_weight': sw_batches})
+
+    # Check on the indices properties
+    assert target.control_levels is None
+    assert isinstance(target.sensitive_levels, list)
+    assert (target.sensitive_levels == ['My feature'])
+
+    # Check we have correct return types
+    assert isinstance(target.overall, float)
+    assert isinstance(target.by_group, pd.Series)
+
+    # Check we have expected number of elements
+    assert len(target.by_group) == 2
```

Comment:
Given that the non-streaming tests already do this sort of checking, would it be easier to just put the data into a non-streaming `MetricFrame` and then check for equality?

Reply:
Right!